guidelines.md

```diff
@@ -0,0 +1,91 @@
+
+```$ git checkout master```
+```$ git fetch upstream```
+```$ git merge upstream/master```
+```$ git push origin master```
+The last command pushes the latest code to your forked repository on Github. The origin is the remote pointing to your forked repository on github.
+
+### CREATE A NEW BRANCH FOR A FEATURE OR BUGFIX -
+***
+Normally, all repositories have a master branch which is considered to remain stable and all new features should be made in a separate branch and after completion merged into master branch. So we should create a new branch for our feature or bugfix and start working on the issue.
+
+```$ git checkout -b <feature-branch>```
+This will create a new branch out of master branch. Now start working on the problem and commit your changes.
+
+```$ git add --all```
+```$ git commit -m "<commit message>"```
+The first command adds all the files or you can add specific files by removing -a and adding the file names. The second command gives a message to your changes so you can know in future what changes this commit makes. If you are solving an issue on original repository, you should add the issue number like #35 to your commit message. This will show the reference to commits in the issue.
+
+### REBASE YOUR FEATURE BRANCH WITH UPSTREAM-
+***
+It can happen that your feature takes time to complete and other contributors are constantly pushing code. After completing the feature your feature branch should be rebase on latest changes to upstream master branch.
+
+```$ git checkout <feature-branch>```
+```$ git pull --rebase upstream master```
+Now you get the latest commits from other contributors and check that your commits are compatible with the new commits. If there are any conflicts solve them.
+
+### SQUASHING YOUR COMMITS-
+***
+You have completed the feature, but you have made a number of commits which make less sense. You should squash your commits to make good commits.
+
+```$ git rebase -i HEAD~5```
+This will open an editor which will allow you to squash the commits.
+
+### PUSH CODE AND CREATE A PULL REQUEST -
+***
+Till this point you have a new branch with the feature or bugfix you want in the project you had forked. Now push your new branch to your remote fork on github.
+
+```$ git push origin <feature-branch>```
+Now you are ready to help the project by opening a pull request means you now tell the project managers to add the feature or bugfix to original repository. You can open a pull request by clicking on green icon -
+
+<p align="center">  <img  src="https://i.imgur.com/aGaqAD5.png">  </p>
+
+Remember your upstream base branch should be master and source should be your feature branch. Click on create pull request and add a name to your pull request. You can also describe your feature.
+
+Awesome! You have made your first contribution. If you have any doubts please let me know in the comments.
```

Comment:
You can remove the line "If you have any doubts please let me know in the comments"